org.opentosca.planbuilder/src/org/opentosca/planbuilder/AbstractTransformingPlanbuilder.java

```diff
@@ -291,44 +446,70 @@ private boolean isCommonSubgraph(Set<AbstractNodeTemplate> graph1, Set<AbstractN
             boolean matchedIn2 = false;
 
             for (AbstractNodeTemplate nodeIn1 : graph1) {
-                if (this.mappingEquals(nodeIn1, nodeTemplate)) {
+                if (this.mappingEquals(nodeTemplate,nodeIn1)) {
                     matchedIn1 = true;
                     break;
                 }
             }
 
             for (AbstractNodeTemplate nodeIn2 : graph2) {
-                if (this.mappingEquals(nodeIn2, nodeTemplate)) {
+                if (this.mappingEquals(nodeTemplate, nodeIn2)) {
                     matchedIn2 = true;
                     break;
                 }
             }
 
-            if (!matchedIn1) {
-                return false;
-            }
-
-            if (!matchedIn2) {
+            if (!matchedIn1 | !matchedIn2) {
```

Comment:
short-circuit? Also personally I'd prefer the de-morganed `!(matchedIn1 && matchedIn2)`, but YMMV